packages/web/postInstall.js

```diff
@@ -1,6 +1,66 @@
 // this copies the examples folder from the @jscad/examples repository locally
 const path = require('path')
-const examplesPath = path.resolve('./node_modules/@jscad/examples')
+const examplesSrc = path.resolve('./node_modules/@jscad/examples')
 const copydir = require('copy-dir')
 const fs = require('fs')
-if (fs.existsSync(examplesPath)) copydir.sync(examplesPath, 'examples')
+const DocBlock = require('docblock')
+const docBlock = new DocBlock()
+
+const copyAndProcessExamples = (examplesSrc) => {
+  const examples = { 'Creating Shapes': [], 'Manipulating Shapes': [], 'Colors': [], 'Parameters': [], 'Imports': [], 'Other': [] }
+  const examplesDist = 'examples'
+  if (fs.existsSync(examplesSrc)) {
+    console.log('Copying Examples...', examplesSrc)
+    copydir.sync(examplesSrc, examplesDist, { mode: false })
+    processExamplesInDirectory(examplesDist, examples)
+    sortExamples(examples)
+    fs.writeFile('examples/examples.json', JSON.stringify(examples), function (err) {
+      if (err) return console.log(err)
+    })
+  } else {
+    return console.log('Examples directory does not exist: ' + examplesSrc)
+  }
+}
+
+const processExamplesInDirectory = (dir, examples) => {
```

Comment:
Hmm... just thinking about this a little further.. can the generation of the JSON file become a function of the examples? possibly generated as part of postinstall? and committed to GIT as well?

The reason for this is easy distribution of a release. It would be great to keep the distribution as easy as 'copy from here to there'.

Reply:
We could, but I'm not sure I agree that we should. 

Moving the generation to postinstall.js of the examples is reasonable, however, examples.json doesn't support the examples, it supports the ui.  If we called it examplesMenuData.json, it clearly belongs where it is.

Also committing something that was autogenerated to git is a bit of a smell. If I had untracked changes after bootstrap, I'd assume the devs forgot to add something to .gitignore.

Reply:
Hmm... OK.

Let’s see what can be done to autogenerate and distribute changes. I was hoping for an easy solution but even the API documentation needs to be generated every release. So, maybe there are some additional steps.